File: 686.重复叠加字符串匹配.py
#
# @lc app=leetcode.cn id=686 lang=python3
#
# [686] 重复叠加字符串匹配
#

# @lc code=start
import math


class Solution:
    def repeatedStringMatch(self, A: str, B: str) -> int:
        nums = math.ceil(len(B) / len(A))
        tmp = A * nums
        if tmp.find(B) != -1:
            return nums
        tmp += A
        return -1 if tmp.find(B) == -1 else nums + 1

# @lc code=end

File: 686.重复叠加字符串匹配.java
/*
 * @lc app=leetcode.cn id=686 lang=java
 *
 * [686] 重复叠加字符串匹配
 */

// @lc code=start
class Solution {
    public int repeatedStringMatch(String A, String B) {
        int num = (int) Math.ceil(B.length()/(A.length()+0.0));
        StringBuilder builder = new StringBuilder();
        for (int i = 0; i < num; i++) {
            builder.append(A);
        }
        if (builder.indexOf(B) != -1) 
            return num;
        builder.append(A);
        return builder.indexOf(B) != -1 ? num+1 : -1;

    }
}
// @lc code=end

